src/export/InstanceExporter.ts

```diff
@@ -202,8 +202,8 @@ export class InstanceExporter implements Fishable {
             sourceInfo: rule.sourceInfo
           });
           // Check if the rule we just validated was at a valid contained path to keep track for resolving Canonical references
-          // It doesn't matter the index/slice name/child path, so just check the first part of the path is 'contained'
-          if (/^contained(?:$|[[.])/.test(rule.path)) {
+          // Only check if the rule was a directly contained resources (aka 'contained' or 'contained.name/url/id', optionally with slice names or indices)
+          if (/^contained(\[([^\]]+)\](\[\d+\])?)?((\.url|\.name|\.id))?$/.test(rule.path)) {
```

Comment:
This regex looks correct, but it may be possible to simplify it a little bit. I think that `^contained(\[[^\]]+\])?(\[\d+\])?(\.url|\.name|\.id)?$` should be equivalent? Also, the capture group for the slice name could be stricter based on what characters are actually allowed for slice names, but since we've already validated the rule, I feel like we probably don't need that here.

Reply:
I like @mint-thompson's simplification, but if `contained` is sliced and then resliced, you could have _multiple_ slice components, so that first `?` really ought to be `*`:
```
/^contained(\[[^\]]+\])*(\[\d+\])?(\.url|\.name|\.id)?$/
```

Since we're not really _validating_ here, however, I think we could be even _looser_ since a numeric index matches the named slice expression anyway:
```
/^contained(\[[^\]]+\])*(\.url|\.name|\.id)?$/
```
Right? `contained`, optionally followed by any number of bracketed tokens and/or optionally followed by url, name, or id subproperty.

Reply:
Yep! It checks out to me and the tests pass. [29f5bf0](https://github.com/FHIR/sushi/pull/1405/commits/29f5bf0515f6ee24f5e14390039345ddaff58750) has the update. I'm also running another regression to just make sure nothing pops up there.